uaclient/tests/test_cli_status.py

```diff
@@ -0,0 +1,97 @@
+import mock
+import socket
+
+import pytest
+
+from uaclient.testing.fakes import FakeConfig
+from uaclient import util
+
+try:
+    from typing import Any, Dict, Optional  # noqa: F401
+except ImportError:
+    # typing isn't available on trusty, so ignore its absence
+    pass
+
+from uaclient.cli import action_status
+
+M_PATH = "uaclient.cli."
+
+
+RESPONSE_LIVEPATCH_AVAILABLE = [{"name": "livepatch", "available": True}]
+UNATTACHED_STATUS = """\
+SERVICE       AVAILABLE  DESCRIPTION
+livepatch     yes        Canonical Livepatch service
+
+This machine is not attached to a UA subscription.
+See https://ubuntu.com/advantage
+"""
+
+ATTACHED_STATUS = """\
+SERVICE       ENTITLED  STATUS    DESCRIPTION
+cc-eal        no        —         Common Criteria EAL2 Provisioning Packages
+cis-audit     no        —         Center for Internet Security Audit Tools
+esm-infra     no        —         UA Infra: Extended Security Maintenance
+fips          no        —         NIST-certified FIPS modules
+fips-updates  no        —         Uncertified security updates to FIPS modules
+livepatch     no        —         Canonical Livepatch service
+
+Enable services with: ua enable <service>
+
+                Account: test_account
+           Subscription: test_contract
+            Valid until: n/a
+Technical support level: n/a
+"""
+
+
+@mock.patch(
+    M_PATH + "contract.get_available_resources",
+    return_value=RESPONSE_LIVEPATCH_AVAILABLE,
+)
+@mock.patch(M_PATH + "os.getuid", return_value=0)
+class TestActionStatus:
+    def test_attached(self, m_getuid, m_get_avail_resources, capsys):
+        """Check that root and non-root will emit attached status"""
+        cfg = FakeConfig.for_attached_machine()
+        assert 0 == action_status(mock.MagicMock(), cfg)
+        # capsys already converts colorized non-printable chars to space
+        # String non-printables from output
+        printable_stdout = capsys.readouterr()[0].replace(
+            "                 ", "        "
+        )
+        assert ATTACHED_STATUS == printable_stdout
+
+        m_getuid.return_value = 1000
+        assert 0 == action_status(mock.MagicMock(), cfg)
+        # capsys already converts colorized non-printable chars to space
+        # String non-printables from output
+        printable_stdout = capsys.readouterr()[0].replace(
+            "                 ", "        "
+        )
+        assert ATTACHED_STATUS == printable_stdout
+
+    def test_unattached(self, m_getuid, m_get_avail_resources, capsys):
+        """Check that root and non-root emit unattached status"""
+        cfg = FakeConfig()
+
+        assert 0 == action_status(mock.MagicMock(), cfg)
+        assert UNATTACHED_STATUS == capsys.readouterr()[0]
+        m_getuid.return_value = 1000
```

Comment:
This isn't necessarily an effective test for non-root; you may also need to instantiate a new config (because its in-memory caches are in a state we can never reach as non-root in production).

Reply:
I should have added that comment, but just wanted to make sure plumbing worked for root and non-root status, which is actually tested a level below in uaclient/tests/test_config:446 with test_nonroot_unattached_is_same_as_unattached_root.
So no need to duplicate that test up at this level as we've already unit tested root vs non-root plumbing.

Reply:
I've dropped the non-root part of this test.